### PR TITLE
Parse virus error message

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
Enhances existing xml parser to support exception and message.
Detects exception from anti-virus app.

Needed for: https://github.com/nextcloud/android/pull/2353

Signed-off-by: tobiasKaminsky <tobias@kaminsky.me>

### CODE REVIEW

File: src/com/owncloud/android/lib/common/operations/ExceptionParser.java [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+     * @throws XmlPullParserException
+     * @throws IOException
+     */
+    private void skip(XmlPullParser parser) throws XmlPullParserException, IOException {
+        if (parser.getEventType() != XmlPullParser.START_TAG) {
+            throw new IllegalStateException();
+        }
+        int depth = 1;
+        while (depth != 0) {
+            switch (parser.next()) {
```
[the remark this answers is not in the document]
Comment (tobiasKaminsky) [reply src/com/owncloud/android/lib/common/operations/ExceptionParser.java]: I did not changed anything on this code part and I do not know any good/sane default.

**Comment by AndyScherzinger** [comment]
:+1: Nice :)

**Comment by tobiasKaminsky** [comment]
@mario ping :-)

**Comment by mario** [comment]
We need to change the name "Virus" to "Malware really".

**Comment by tobiasKaminsky** [comment]
Is it called Virus as the app is named virus and the exception is also naming it virus, so I thought it is better to stay in sync.
In general I like your idea, but I prefer to then change it also/first on server side.

**Comment by mario** [comment]
@tobiasKaminsky link to the server app?

**Comment by mario** [comment]
👍 
Code wise fine, with the goal to change this to malware.

**Pull request merged**